src/pages/students/edit-student/EditStudent.tsx

```diff
@@ -0,0 +1,135 @@
+import StudentData, { type StudentDataProps } from "../forms/studentDataForm/StudentData"
+import PaymentData, { type PaymentDataProps } from "../forms/paymentDataForm/PaymentData"
+import { useGetStudentByIdQuery, useUpdateStudentMutation } from "../../../app/services/StudentService"
+import useAuthentication from "../../../hooks/useAuthentication";
+import { useRef, type JSX } from "react";
+//import PlanData, { type PlanDataProps } from "../forms/planDataForm/PlanData";
+import { useLocation, useNavigate } from "react-router";
+import { useParams } from "react-router-dom";
+import { MainContainer, TitleContainer, Title, ButtonsContainer, Button, FormContainer } from "./EditStudent.styles";
+import { formatDateToDash, formatDateToISO } from "../../../utils/DateFormatter";
+import toast from "react-hot-toast";
+import type { UpdateStudentRequest } from "../../../app/types/requests/UpdateStudentRequest.type";
+import BackIcon from "../../../assets/back-icon.svg";
+
+
+function EditStudent() {
+
+    const { userId } = useAuthentication();
+
+    const formStudentDataRef = useRef<any>(undefined);
+
+    const formPaymentDataRef = useRef<any>(undefined);
+
+    const { numberOfStep } = useParams();
+
+    const { studentId } = useLocation().state;
+
+    const numberOfStepNum = numberOfStep ? parseInt(numberOfStep) : null;
+
+    const { data: studentSaveData } = useGetStudentByIdQuery(studentId!);
+
+    const [updateStudent] = useUpdateStudentMutation();
+
+    const navigate = useNavigate();
+
+    const showStudentData = (studentSaveData: StudentDataProps) => {
+
+        const formattedBirthday = formatDateToISO(studentSaveData.birthday);
+
+        return {
+            name: studentSaveData.name,
+            lastName: studentSaveData.lastName,
+            dni: studentSaveData.dni,
+            cellphoneNumber: studentSaveData.cellphoneNumber,
+            birthday: formattedBirthday,
+            pathologies: studentSaveData.pathologies ?? undefined,
+        };
+    }
+
+    if (!studentSaveData) {
+        return <p>Cargando datos del estudiante...</p>;
+    }
```

Comment:
Aca no tendriamos que usar este condicional, sino aprovechar los otros datos que te devuelve RTK Query como el isLoading o el isError

Si hicieramos esto asi

```
    const { data: studentSaveData, isLoading, isError } = useGetStudentByIdQuery(studentId!);
```

aca podriamos usar el isLoading


```suggestion
    if (isLoading) {
        return <p>Cargando datos del estudiante...</p>;
    }
```

y tener otro condicional

```
    if (isError || !studentSaveData) {
        return <p>Hubo un error al obtener la informacion del estudiante. Intente nuevamente</p>;
    }
```

Reply:
Otra cosa es que podriamos tener un loader

<img width="441" height="291" alt="Image" src="https://github.com/user-attachments/assets/0fc698a6-9527-441a-8a3b-132c4434a4e9" />